.teamcity/src/main/kotlin/configurations/FlakyTestQuarantine.kt

```diff
@@ -0,0 +1,47 @@
+package configurations
+
+import common.BuildToolBuildJvm
+import common.Os
+import common.applyDefaultSettings
+import common.buildToolGradleParameters
+import common.checkCleanM2AndAndroidUserHome
+import common.functionalTestExtraParameters
+import common.functionalTestParameters
+import common.gradleWrapper
+import common.killProcessStep
+import jetbrains.buildServer.configs.kotlin.v2019_2.BuildStep
+import model.CIBuildModel
+import model.Stage
+
+class FlakyTestQuarantine(model: CIBuildModel, stage: Stage, os: Os) : BaseGradleBuildType(stage = stage, init = {
+    id("${model.projectId}_FlakyQuarantine_${os.name.capitalize()}")
+    name = "Flaky Test Quarantine - ${os.name.capitalize()}"
+    description = "Run all flaky tests skipped multiple times"
+
+    applyDefaultSettings(os, BuildToolBuildJvm, 60)
+
+    val testsWithOs = model.stages.flatMap { it.functionalTests }.filter { it.os == os }
+
+    testsWithOs.forEach { testCoverage ->
+        val extraParameters = functionalTestExtraParameters("FlakyTestQuarantine", os, testCoverage.testJvmVersion.major.toString(), testCoverage.vendor.name)
+        val parameters = (
+            buildToolGradleParameters(true) +
+                listOf("-PflakyTestQuarantine") +
+                listOf(extraParameters) +
+                functionalTestParameters(os)
+            ).joinToString(separator = " ")
+        steps {
```

Comment:
So we'd run one step for each type of test coverage? How many steps do we end up here?

And I suppose we run those steps without test distribution. right?

Reply:
I managed to generate TC config with old version of `pom`. There will be 45 steps (including ReadyForRelease) or 33 steps (not including ReadyForRelease)

Reply:
After excluding experimental stages, there're only 9 steps. which seems reasonable. 

Yes, it's without TD, let's see how it works.